vms/platformvm/blocks/builder/builder.go

```diff
@@ -147,8 +147,12 @@ func (b *builder) AddUnverifiedTx(tx *txs.Tx) error {
 		return err
 	}
 
-	if err := b.Mempool.Add(tx); err != nil {
-		return err
+	// If we are lite syncing the Primary Network, we should not be maintaining
+	// the transaction mempool locally.
```

Comment:
Wondering if it would be easier to maintain if we used a noop mempool implementation rather than adding the check here + in rejection.

Reply:
To connect this to your other comment (https://github.com/ava-labs/avalanchego/pull/1769/files#r1287805590), I think that making this a no-op won't be enough to avoid passing this through the VM.

I think we'll want to avoid doing almost anything over the network as well (other than gossiping our txs from the RPC, but we shouldn't accept incoming gossip).

Reply:
yeah